src/test/regress/sql/multi_hash_pruning.sql

```diff
@@ -27,11 +27,40 @@ SET client_min_messages TO DEBUG2;
 -- Check that we can prune shards for simple cases, boolean expressions and
 -- immutable functions.
 
+
+-- Since router plans are not triggered for task-tracker executor type,
+-- we need to run the tests that triggers router planning seperately for
+-- both executors. Otherwise, check-full fails on the task-tracker.
+-- Later, we need to switch back to the actual task executor
+-- to contuinue with correct executor type for check-full.
+SELECT quote_literal(setting) AS actual_task_executor
```

Comment:
This feels a bit like overkill. Is it really critical that we run the remaining queries with both real-time and task-tracker?

Anyway, better to use SELECT current_setting('citus.task_executor_type')